src/sentry/seer/endpoints/organization_seer_onboarding.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+import logging
+
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import OrganizationEndpoint, OrganizationPermission
+from sentry.api.serializers.rest_framework import CamelSnakeSerializer
+from sentry.models.organization import Organization
+from sentry.seer.autofix.autofix import onboarding_seer_settings_update
+from sentry.seer.endpoints.project_seer_preferences import BranchOverrideSerializer
+from sentry.seer.models import SeerRepoDefinition
+
+logger = logging.getLogger(__name__)
+
+
+class RepositorySerializer(CamelSnakeSerializer):
+    provider = serializers.CharField(required=True)
+    owner = serializers.CharField(required=True)
+    name = serializers.CharField(required=True)
+    external_id = serializers.CharField(required=True)
+    organization_id = serializers.IntegerField(required=False, allow_null=True)
+    integration_id = serializers.CharField(required=False, allow_null=True)
+    branch_name = serializers.CharField(required=False, allow_null=True)
+    branch_overrides = BranchOverrideSerializer(many=True, required=False, default=list)
+    instructions = serializers.CharField(required=False, allow_null=True, allow_blank=True)
+    base_commit_sha = serializers.CharField(required=False, allow_null=True)
+    provider_raw = serializers.CharField(required=False, allow_null=True)
+
+
+class ProjectRepoMappingField(serializers.Field):
+    def to_internal_value(self, data):
+        if not isinstance(data, dict):
+            raise serializers.ValidationError("Expected a dictionary")
+
+        result = {}
+        for project_id_str, repos_data in data.items():
```

Comment:
are we validating that these project_ids belong to the requesting organization somewhere? would be good to add a test to make sure one org can't change another project's project settings